mmd_tools/operators/morph.py

```diff
@@ -226,13 +234,23 @@ def execute(self, context):
         rig = mmd_model.Model(root)
         mmd_root = root.mmd_root
         meshObj = rig.firstMesh()
-        if meshObj is None:
-            self.report({ 'ERROR' }, "The model mesh can't be found")
-            return { 'CANCELLED' }
+
         morph = mmd_root.material_morphs[mmd_root.active_morph]
         if len(morph.data) == 0:
             return { 'FINISHED' }
         mat_data = morph.data[morph.active_material_data]
+
+        # if mmd_root.advanced_mode:
+        relMesh = rig.findMesh(mat_data.related_mesh)
+        if relMesh is not None:
+            meshObj = relMesh
+        else:
+            self.report({ 'ERROR' }, "The related mesh can't be found")
+            return { 'CANCELLED' }
+        
+        if meshObj is None:
+            self.report({ 'ERROR' }, "The model mesh can't be found")
+            return { 'CANCELLED' }        
         work_mat_name = mat_data.material+"_temp"
```

Comment:
I came from a Java background so I like the things precise and explicit. And, while to the user will not make a difference, technically "Not finding the model mesh" is not the same as "Not finding the related mesh". The former means we can't find any mesh while the later refers only to the mesh only to the mesh related with the active morph data. But you can change it as you like.